### ISSUE TITLE
Bundle Items price after saving to 0.00 go back to default price value Bundle Product Edit

### ISSUE THREAD

**Issue by jeredmilla** [magento/magento2#32383]
### Preconditions (*)
1. Magento 2.4.2
1. Magento 2.4-develop

### Steps to reproduce (*)
1. Make a Bundle Product on the admin dashboard
1. Set the Bundle Items price to zero (0.00)
1. Save it
1. Go back again to the Bundle Items price you will see the price is not zero it will display the child product price.

### Expected result (*)
1. Saving it with Bundle Items zero price when you go back to the Bundle product edit it should be zero price. 
![image](https://user-images.githubusercontent.com/56057608/110400188-29473a80-80b2-11eb-8c0c-84842d58fa52.png)

### Actual result (*)
1. Save the Bundle Product with bundle items price zero.
![image](https://user-images.githubusercontent.com/56057608/110400259-4bd95380-80b2-11eb-85c5-49e27a7022b9.png)

1. When you go back to the Bundle Product Edit the bundle item price displayed the item default price.
![image](https://user-images.githubusercontent.com/56057608/110400369-7b885b80-80b2-11eb-95da-004cbdc5f51a.png)

### Testing notes

The case from https://github.com/magento/magento2/issues/28431 needs to be tested as well during verification of this issue fix

## 
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- < ] Severity: **S0** *- Affects critical data or functionality and leaves users without workaround.*
- [x] Severity: **S1** *- Affects critical data or functionality and forces users to employ a workaround.*
- [ ] Severity: **S2** *- Affects non-critical data or functionality and forces users to employ a workaround.*
- [ ] Severity: **S3** *- Affects non-critical data or functionality and does not force users to employ a workaround.*
- [ > Severity: **S4** *- Affects aesthetics, professional look and feel, “quality” or “usability”.*

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Bravo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-41177` were created

**Issue Available**: @engcom-Bravo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by JoostWan** [comment]
@jeredmilla We have the same issue. Did some debugging and the issue has been created since this commit https://github.com/magento/magento2/commit/8974e16d98b6154c64b5de6288d3d7ac7cc7291e. 

```
if ($bundledProductPrice <= 0){
    $bundledProductPrice = $selection->getPrice();
}
```

**Comment by sivaschenko** [comment]
The case from https://github.com/magento/magento2/issues/28431 needs to be tested as well during verification of this issue fix

**Comment by jeredmilla** [comment]
@JoostWan Thanks it is working when removing this.
@sivaschenko I will wait for the update for this. We are not still upgrading to 2.4.2. Thanks!

**Comment by magento-engcom-team** [comment]
Hi @jeredmilla, @engcom-Kilo. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-41177` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.
